File: 0135-candy/0135-candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        if (len(ratings) <= 1):
            return len(ratings)

        candies = []
        for i in range (len(ratings)):
            candies.append(1)

        # From front
        for i in range (1, len(ratings)):
            if (ratings[i-1] < ratings[i] and candies[i-1] >= candies[i]):
                candies[i] = candies[i-1] + 1

        # From back
        for i in range ((len(ratings)-1) -1, -1, -1):
            if (ratings[i+1] < ratings[i] and candies[i+1] >= candies[i]):
                candies[i] = candies[i+1] + 1

        total_candies = 0
        for candy in candies:
            total_candies += candy
        
        print(candies)
        

        return total_candies
            

File: 0135-candy/0135-candy.java
class Solution {
    public int candy(int[] ratings) {
        int n = ratings.length;
        int total = n;

        int[] additionalCandies = new int[n];
        
        // Left to Right
        additionalCandies[0] = 0;
        for (int i = 1; i < n; i++){
            if (ratings[i] > ratings[i-1]){
                additionalCandies[i] = additionalCandies[i-1] + 1;
            } 
        }

        // Right to Left
        additionalCandies[0] = 0;
        for (int i = n-2; i >= 0; i--){
            if (ratings[i] > ratings[i+1] && 
                additionalCandies[i] <= additionalCandies[i+1]
            ){
                additionalCandies[i] = additionalCandies[i+1] + 1;
            } 
        }
        for (int candy : additionalCandies ){
            total += candy;
        }
        return total;
    }
}